### ISSUE TITLE
no-unused-vars false positives on called back functions

### ISSUE THREAD

**Issue by Wiebelt** [eslint/eslint#7317]
**Tell us about your environment**

* **ESLint Version: 3.0.0**
* **Node Version: 4.5.0**
* **npm Version: 2.15.9**

**What parser (default, Babel-ESLint, etc.) are you using?**
The default parser

**Please show your full configuration:**

```yaml
---
env:
  browser: true
globals: {}
rules:
  no-unused-vars:
    - error
    - vars: local
      args: none
```

**What did you do? Please include the actual source code causing the issue.**

```js
function test() {
	var myFunc;
	myFunc = setTimeout(function () {
		console.info(myFunc);
	}, 6000);
}
test();
```

**What did you expect to happen?**
The code should produce no errors.

**What actually happened? Please include the actual, raw output from ESLint.**
Eslint gives the following error:
```
  3:6  error  'myFunc' is defined but never used  no-unused-vars
```

When i remove the test function around it it does pass. As in:
```js
	var myFunc;
	myFunc = setTimeout(function () {
		console.info(myFunc);
	}, 6000);
```

**Comment by platinumazure** [comment]
Definitely seems inconsistent- probably a bug. Marking as "evaluating" since I can't reproduce on this device.

**Comment by not-an-aardvark** [comment]
I can't reproduce this issue in the demo:

![](https://i.gyazo.com/94b5d9d3c8cd4cad90978b30b23617b0.png)

Could you try upgrading to the latest version of ESLint (3.7.1 at the moment) to see if the issue still occurs?

**Comment by Wiebelt** [comment]
Aah, yes. Upgrading to 3.7.1 fixed it, i should have thought of that!. Thanks :-)
